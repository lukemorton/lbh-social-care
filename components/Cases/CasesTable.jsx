import PropTypes from 'prop-types';
import { formatDate } from 'utils/date';
const onClick = (url) => window.open(url, '_blank');

const CasesEntry = ({
  formName,
  caseFormUrl,
  officerEmail,
  caseFormTimestamp,
}) => (
  <tr className="govuk-table__row">
    <td className="govuk-table__cell govuk--timestamp">
      {caseFormTimestamp && formatDate(caseFormTimestamp)}{' '}
    </td>
    <td className="govuk-table__cell">{formName}</td>
    <td className="govuk-table__cell">
      {officerEmail && `- created by ${officerEmail}`}
    </td>
    <td className="govuk-table__cell govuk-button--secondary'">
      <a
        href="#"
        className="govuk-link govuk-custom-text-color"
        onClick={() => onClick(caseFormUrl)}
      >
        View
      </a>
    </td>
  </tr>
);

const CasesTable = ({ records }) => (
  <table className="govuk-table">
    <tbody className="govuk-table__body">
      {records.map((result) => (
        <CasesEntry key={result.personId} {...result} />
      ))}
    </tbody>
  </table>
);

CasesTable.propTypes = {
  records: PropTypes.arrayOf(
    PropTypes.shape({
      personId: PropTypes.number.isRequired,
      formName: PropTypes.string.isRequired,
      caseFormUrl: PropTypes.string.isRequired,
      officerEmail: PropTypes.string.isRequired,
      caseFormTimestamp: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default CasesTable;
