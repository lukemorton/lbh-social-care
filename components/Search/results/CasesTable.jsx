import PropTypes from 'prop-types';

import { formatDate, isDateValid } from 'utils/date';

const onClick = (url) => window.open(url, '_blank');

const CasesEntry = ({
  firstName,
  lastName,
  caseFormUrl,
  dateOfBirth,
  officerEmail,
  caseFormTimestamp,
}) => (
  <tr className="govuk-table__row">
    <td className="govuk-table__cell">
      {firstName} {lastName}
    </td>
    <td className="govuk-table__cell">
      {isDateValid(dateOfBirth) && dateOfBirth}
    </td>
    <td className="govuk-table__cell">{officerEmail}</td>
    <td className="govuk-table__cell">{formatDate(caseFormTimestamp)}</td>
    <td className="govuk-table__cell govuk-button--secondary'">
      <a
        href="#"
        className="govuk-link govuk-custom-text-color"
        onClick={() => onClick(caseFormUrl)}
      >
        View
      </a>
    </td>
  </tr>
);

const CasesTable = ({ records }) => (
  <table className="govuk-table">
    <thead className="govuk-table__head">
      <tr className="govuk-table__row">
        <th scope="col" className="govuk-table__header">
          Client Name
        </th>
        <th scope="col" className="govuk-table__header">
          Date of birth
        </th>
        <th scope="col" className="govuk-table__header">
          Uploaded by
        </th>
        <th scope="col" className="govuk-table__header">
          Last upload
        </th>
        <th scope="col" className="govuk-table__header"></th>
      </tr>
    </thead>
    <tbody className="govuk-table__body">
      {records.map((result) => (
        <CasesEntry key={result.personId} {...result} />
      ))}
    </tbody>
  </table>
);

CasesTable.propTypes = {
  records: PropTypes.arrayOf(
    PropTypes.shape({
      personId: PropTypes.number.isRequired,
      formName: PropTypes.string.isRequired,
      caseFormUrl: PropTypes.string.isRequired,
      officerEmail: PropTypes.string.isRequired,
      caseFormTimestamp: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default CasesTable;
