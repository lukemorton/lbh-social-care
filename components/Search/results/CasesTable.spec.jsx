import { render } from '@testing-library/react';

import CasesTable from './CasesTable';

describe('CasesTable component', () => {
  const props = {
    records: [
      {
        personId: 123,
        dateOfBirth: '25/10/2000',
        caseFormTimestamp: '25/10/2020 13:49:43',
        firstName: 'foo',
        lastName: 'bar',
        caseFormUrl: 'https://foo.bar',
        officerEmail: 'Fname.Lname@hackney.gov.uk',
        formName: 'foo',
      },
    ],
  };
  it('should render properly', () => {
    const { asFragment } = render(<CasesTable {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
