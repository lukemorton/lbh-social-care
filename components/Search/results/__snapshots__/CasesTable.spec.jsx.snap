// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CasesTable component should render properly 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Client Name
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Date of birth
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Uploaded by
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Last upload
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell"
        >
          foo bar
        </td>
        <td
          class="govuk-table__cell"
        >
          25/10/2000
        </td>
        <td
          class="govuk-table__cell"
        >
          Fname.Lname@hackney.gov.uk
        </td>
        <td
          class="govuk-table__cell"
        >
          Oct 25, 2020
        </td>
        <td
          class="govuk-table__cell govuk-button--secondary'"
        >
          <a
            class="govuk-link govuk-custom-text-color"
            href="#"
          >
            View
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
